sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1202,11 +1203,16 @@ class Analyzer(
           if (resolvedOperator.resolved) {
             // Try to replace all aggregate expressions in the filter by an alias.
             val aggregateExpressions = ArrayBuffer.empty[NamedExpression]
-            val transformedAggregateFilter = resolvedAggregateFilter.transform {
+            val transformedAggregateFilter = resolvedAggregateFilter.transformDown {
               case ae: AggregateExpression =>
                 val alias = Alias(ae, ae.toString)()
                 aggregateExpressions += alias
                 alias.toAttribute
+              case ne: NamedExpression => ne
+              case e: Expression =>
+                val alias = Alias(e, e.toString)()
+                aggregateExpressions += alias
+                alias.toAttribute
```

Comment:
looks like we are blindly pushing all expression into `Aggregate`, how about we just push the whole filter condition?